.github/workflows/deploy-docs.yml

```diff
@@ -8,26 +8,95 @@ on:
     paths:
       - '.github/workflows/deploy-docs.yml'
       - 'docs/**'
+    # NOTE: The tag push event is not constrained by branch and path `on` conditions.
+    # For taking a snapshot of the docs state from the tagged commit,
+    # to a versioned subdirectory with dir name matching the git tag substring `v<MAJOR>.<MINOR>`:
+    tags:
+      - 'v[0-9]+.[0-9]+*'
 
-# Jobs will run shell commands from this subdirectory:
-defaults:
-  run:
-    working-directory: docs
+env:
+  # Default docs version to build and deploy
+  DOCS_VERSION: edge
+  GIT_USER: 'github-actions[bot]'
+  GIT_EMAIL: '41898282+github-actions[bot]@users.noreply.github.com'
 
 jobs:
+  add-version-to-docs:
+    if: startsWith(github.ref, 'refs/tags/')
+    name: 'Handle `versions.json` in `gh-pages` branch'
```

Comment:
Back-Ticks are not rendered in Actions menu. We should remove the ``` ` ```s

Reply:
They still provide some context. You can mix quotes if you like? Or escape single quotes. Even though it's not rendered, I think the intent of backticks in the string names is still clear.